src/Steam.Common/SteamId.cs

```diff
@@ -0,0 +1,492 @@
+        }
+
+        /// <summary>
+        /// Creates an anonymous user SteamID in the specified universe
+        /// </summary>
+        /// <param name="universe">The universe to set</param>
+        public static SteamId CreateAnonymousUser(Universe universe)
+        {
+            return new SteamId(0, AccountType.AnonUser, universe, 0);
+        }
+
+        /// <summary>
+        /// Creates an anonymous game server SteamID in the specified universe
+        /// </summary>
+        /// <param name="universe">The universe to set</param>
+        public static SteamId CreateAnonymousGameServer(Universe universe)
+        {
+            return new SteamId(0, AccountType.AnonGameServer, universe, 0);
+        }
+
+        /// <summary>
+        /// Creates an anonymous individual account SteamID with the specified ID, universe, and instance flags
+        /// </summary>
+        /// <param name="id">The account ID</param>
+        /// <param name="universe">The universe</param>
+        /// <param name="instance">The instance flags</param>
+        public static SteamId CreateIndividualAccount(uint id, Universe universe, Instance instance)
+        {
+            return new SteamId(id, AccountType.Individual, universe, (uint)instance);
+        }
+
+        /// <summary>
+        /// Creates a new ID from the specified ID string in the format STEAM_X:Y:Z
+        /// </summary>
+        /// <param name="id">A string containing the Steam2 ID to parse</param>
+        public static SteamId FromSteam2(string id)
+        {
+            var match = Steam2Regex.Match(id);
+            if (!match.Success)
+                throw new ArgumentException("The input string does not match the Steam2 pattern");
+
+            var universe = (Universe)int.Parse(match.Groups[1].Value);
+            var magic = uint.Parse(match.Groups[2].Value);
+            var account = uint.Parse(match.Groups[3].Value);
+            return new SteamId((account * 2) + magic, AccountType.Individual, universe, 1);
+        }
+
+        /// <summary>
+        /// Creates a new ID from the specified ID string in the format [W:X:Y:Z]
+        /// </summary>
+        /// <param name="id">A string containing the Steam3 ID to parse</param>
+        /// <remarks>
+        /// The instance for returned IDs depends on the input.
+        /// 
+        /// For 'c' and 'L' types, the instance is the ClanFlag or LobbyFlag respectively.
+        /// For 'U' and 'i'/'I' the instance is 1.
+        /// For 'g' and 'T' the instance is 0.
+        /// 
+        /// For all other account types, the instance is Z or 0 if Z wasn't found.
+        /// </remarks>
+        public static SteamId FromSteam3(string id)
+        {
+            var match = Steam3Regex.Match(id);
+            if (!match.Success)
+                throw new ArgumentException("The input string does not match the Steam3 pattern");
+
+            var typeChar = match.Groups[1].Value[0];
```

Comment:
Nice use of groupings, however using named groups might make the code more readable

Reply:
With constant ints?

Reply:
Nah, you can add names to groups in regex. Lke this https://docs.microsoft.com/en-us/dotnet/standard/base-types/grouping-constructs-in-regular-expressions#named-matched-subexpressions